src/app/components/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,61 @@
+import { useNewsContext } from '@/context/news.context';
+import React from 'react';
+import { Selects } from '@/models/types';
+import classNames from 'classnames/bind';
+import getPaginationRange from '@/utils/getPaginationRange';
+import styles from './Pagination.module.scss';
+
+type Props = {
+  currentPage: number;
+  setCurrentPage(value: number): void;
+  selects: Selects;
+  setSelects(selects: Selects): void;
+};
+
+const cx = classNames.bind(styles);
+
+function Pagination({ currentPage, setCurrentPage, selects, setSelects }: Props) {
+  const { news } = useNewsContext();
+  const totalPageNumber = Math.floor(100 / news.length); // 100 articles - API limit
```

Comment:
`100` articles limit is per page. Please use `totalResults` from the api

Reply:
No. It's a limit on the number of articles in general. If you specify pageSize=3 and page=34, you will fail.

Reply:
Completely justified 🙂